src/actions/commands/actions.ts

```diff
@@ -2580,7 +2580,10 @@ class CommandGoToDefinition extends BaseCommand {
     const oldActiveEditor = vimState.editor;
 
     await vscode.commands.executeCommand('editor.action.goToDeclaration');
-
+    // `executeCommand` returns immediately before cursor is updated
+    // wait for the editor to update before updating the vim state
+    // https://github.com/VSCodeVim/Vim/issues/3277
+    await new Promise(resolve => setTimeout(resolve, 1000));
```

Comment:
how about you use getCursorsAfterSync in util.ts and pass in a timeout of 1 second to get a potential boost in performance if it doesnt take a full 1 second

Reply:
Thanks!